remote/remote.go

```diff
@@ -35,7 +35,12 @@ func WithPreviousImage(imageName string) ImageOption {
 	return func(r *Image) (*Image, error) {
 		var err error
 
-		prevImage, err := newV1Image(r.keychain, imageName)
+		currentImagePlatform, err := imagePlatform(r.image)
```

Comment:
same ordering constraints introduced here

Reply:
If we remove `WithDefaultPlatform` for now, then we can have a simpler `fallbackPlatform` and won't need to keep track of the `currentPlatform`